### PR TITLE
Remove OpenSslUtils struct.

### LINKED ISSUE #774: Remove unnecessary OpenSslUtils struct; define methods at namespace level
As mentioned in PR #766, OpenSslUtils struct should not be wrapping the methods currently inside of it. Rather, they should be defined at namespace level.  This TODO tracks doing that small refactor in a subsequent commit.

### PR DESCRIPTION
**Pull request by coryan**
This "struct" only had static member functions. I just moved the private
member functions to the .cc file and things are copacetic.

There is more cleanup that we could do in this file, I am planning to send
those as separate PRs.

This fixes #774.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2141)

### CODE REVIEW

File: google/cloud/storage/internal/openssl_util.h [hunk 1]
```diff
@@ -35,199 +35,67 @@ namespace storage {
...
-  static std::string UrlsafeBase64Encode(
-      std::vector<std::uint8_t> const& bytes) {
-    std::string b64str = Base64Encode(bytes);
-    std::replace(b64str.begin(), b64str.end(), '+', '-');
-    std::replace(b64str.begin(), b64str.end(), '/', '_');
-    RightTrim(b64str, '=');
-    return b64str;
-  }
+/**
+ * Encodes a string using Base64.
```
Comment (devbww) [inline google/cloud/storage/internal/openssl_util.h]: Perhaps "string" is the wrong noun to use here?

File: google/cloud/storage/internal/openssl_util.h [hunk 2]
```diff
@@ -35,199 +35,67 @@ namespace storage {
...
-  };
-#else
-  inline static std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)>
-  GetDigestCtx() {
-    return std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)>(
-        EVP_MD_CTX_new(), &EVP_MD_CTX_free);
-  };
-#endif
-};
+/**
```
Comment (devbww) [inline google/cloud/storage/internal/openssl_util.h]: How does doxygen deal with only having one description for two functions?

File: google/cloud/storage/internal/openssl_util.h [hunk 3]
```diff
@@ -35,199 +35,67 @@ namespace storage {
...
+ * Returns a Base64-encoded version of the given a string, using the URL- and
+ * filesystem-safe alphabet, making these adjustments:
+ * -  Replace '+' with '-'
+ * -  Replace '/' with '_'
+ * -  Right-trim '=' characters
+ */
+inline std::string UrlsafeBase64Encode(std::vector<std::uint8_t> const& bytes) {
```
Comment (devbww) [inline google/cloud/storage/internal/openssl_util.h]: Given that these are inlined in a header, should we replace it with a single template and leave the parameter-type requirement to Base64Encode()?

File: google/cloud/storage/internal/openssl_util.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/openssl_util.h]: Yup. Fixed.

File: google/cloud/storage/internal/openssl_util.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/internal/openssl_util.h]: It does not, it needs a separate comment, thought this is now moot?

File: google/cloud/storage/internal/openssl_util.h (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/internal/openssl_util.h]: Thanks! Much better, fixed.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**